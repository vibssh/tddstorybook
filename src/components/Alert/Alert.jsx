import React from 'react';
import PropTypes from 'prop-types';
import './Alert.css';

const propTypes = {
  children: PropTypes.node,
  type: PropTypes.string
}

const defaultPropTypes = {
  children: null,
  type: ''
}

const Alert = (props) => {
  const {type, children} = props;
  const modifier = type === undefined ? '' : type;
  return (
    <div className={`alert ${modifier}`} role="alert">
      {children}
    </div>
  )
}
Alert.propTypes = propTypes;
Alert.defaultPropTypes = defaultPropTypes;

export default Alert;