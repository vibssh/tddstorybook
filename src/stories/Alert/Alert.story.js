import React from 'react';
import { storiesOf } from '@storybook/react';

import Alert from '../../components/Alert';

const type = {
  DEFAULT: '',
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  SUCCESS: 'success',
  DANGER: 'danger',
  WARNING: 'warning',
  LIGHT: 'light'
}

const iterator = () => {
  const Types = Object.keys(type);
  Types.map((T) => {
    return (
      storiesOf('Alert', module)
        .addWithJSX(`Alert ${type[T]}`, () => (
          <Alert type={type[T]}>A simple default alertâ€”check it out!</Alert>
        )
      )
    )
  })
}

iterator();