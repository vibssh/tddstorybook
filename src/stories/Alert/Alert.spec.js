import React from "react";
import { shallow } from "enzyme";

import Alert from "../../components/Alert";

const Types = {
  DEFAULT: "default",
  PRIMARY: "primary",
  SECONDARY: "secondary",
  SUCCESS: "success",
  DANGER: "danger",
  WARNING: "warning",
  LIGHT: "light"
};

describe("Alert", () => {
  let wrapper;
  it("Should render Default Alert", () => {
    wrapper = shallow(<Alert>It should render Default Alert</Alert>);
    expect(wrapper.debug()).toMatchSnapshot();
  });
  it("Should render Primary Alert", () => {
    wrapper = shallow(<Alert type={Types['PRIMARY']}>It should render Primary Alert</Alert>);
    expect(wrapper.debug()).toMatchSnapshot();
  })
});
