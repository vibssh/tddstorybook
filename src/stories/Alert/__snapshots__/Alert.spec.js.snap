// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Should render Default Alert 1`] = `
"<div className=\\"alert \\" role=\\"alert\\">
  It should render Default Alert
</div>"
`;

exports[`Alert Should render Primary Alert 1`] = `
"<div className=\\"alert primary\\" role=\\"alert\\">
  It should render Primary Alert
</div>"
`;
